typing/env.ml

```diff
@@ -1965,7 +1990,7 @@ and store_label ~check type_decl type_id lbl_id lbl env =
       let used = label_usages () in
       Types.Uid.Tbl.add !used_labels k
         (add_label_usage used);
-      if not (ty_name = "" || ty_name.[0] = '_' || name.[0] = '_')
+      if not (ty_name = "" || ty_name.[0] = '_' || name.[0] = '_' || check_warning_attributes type_kind lbl_id)
```

Comment:
The warning attribute should rather disable the warning as it is done when the attribute is attached to the whole type.